tagstudio/src/qt/widgets/thumb_renderer.py

```diff
@@ -746,6 +757,46 @@ def _video_thumb(self, filepath: Path) -> Image.Image:
             )
         return im
 
+    def _pdf_thumb(self, filepath: Path, size: int) -> Image.Image:
+        file: QFile = QFile(filepath)
+        was_open_success: bool = file.open(
+            QIODeviceBase.OpenModeFlag.ReadOnly, QFileDevice.Permission.ReadUser
+        )
+        if not was_open_success:
+            logging.error(
+                f"[ThumbRenderer][PDF][ERROR]: Couldn't open pdf-file {filepath.name}"
+            )
+            return None
+        document: QPdfDocument = QPdfDocument()
+        document.load(file)
+        # Transform page_size in points to pixels with proper aspect ratio
+        page_size: QSizeF = document.pagePointSize(0)
+        ratio_hw: float = page_size.height() / page_size.width()
+        if ratio_hw >= 1:
+            page_size *= size / page_size.height()
+        else:
+            page_size *= size / page_size.width()
+        # Enlarge image to improve image downscaling (kind of arbitrary)
+        page_size *= 2.5
```

Comment:
Use a good variable name instead; magic numbers like this will be confusing in the future. 
```suggestion
        ENLARGE_SCALE_FACTOR = 2.5
        page_size *= ENLARGE_SCALE_FACTOR 
```

The suggestion is not here for you to use as a drop-in replacement. You should save this to the class and, possibly in the future (or right now), move it to a constant namespace.